### PR TITLE
Move clone_progress to separate list

### PR DESCRIPTION
**Pull request by avlitman**
**What this PR does / why we need it**:
Since clone_progress metric used for internal purpose only we should not document it in metrics.md or test it in the metrics linter. therefore clone_progress moved to separate list (not used by the doc and linter generators).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubevirt/containerized-data-importer/issues/2515
Jira-ticket: https://issues.redhat.com/browse/CNV-33647

**Special notes for your reviewer**:
This fix was discussed with CDI developer.

**Release note**:

```
The "clone_progress" metric has been removed from the project documentation. We have decided to exclude this metric from the documentation as it is intended for internal use only.
```

### CODE REVIEW

File: cmd/cdi-cloner/clone-source.go [hunk 1]
```diff
@@ -96,8 +96,8 @@ func startPrometheus() {
 func createProgressReader(readCloser io.ReadCloser, ownerUID string, totalBytes uint64) io.ReadCloser {
 	progress := prometheus.NewCounterVec(
 		prometheus.CounterOpts{
-			Name: monitoring.MetricOptsList[monitoring.CloneProgress].Name,
```
Comment (sradco) [inline cmd/cdi-cloner/clone-source.go:99]: @avlitman  Why should we keep them here at all?

**Comment by avlitman** [comment]
Please do not merge this PR before @sradco approves.

**Pull request merged**